cypress/integration/channel/message_deletion_spec.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message deletion', () => {
+  before(() => {
+  // 1. Go to Main Channel View with "user-1"
+      cy.toMainChannelView('user-1');
+  });
+
+  it('M13336 Should delete both parent post and reply when deleting parent post from center', () => {
+      // 1. Post message in center.
+      cy.postMessage('test message deletion {enter}');
+
+      cy.getLastPostId().then((parentMessageId) => {
+          // 2. Mouseover the post and click post comment icon.
+          cy.clickPostCommentIcon();
+
+          // * Check that the RHS is open
+          cy.get('#rhsContainer').should('be.visible');
+
+          // 3. Post a reply in RHS.
+          cy.postMessageReplyInRHS('test message reply in RHS {enter}');
+
+          cy.getLastPostId().then((replyMessageId) => {
+              // 4. Click post dot menu in center.
+              cy.clickPostDotMenu(parentMessageId);
+
+              // 5. Click delete button.
+              cy.get(`#CENTER_dropdown_${parentMessageId}`).within(() => {
+                  cy.get('.dropdown-menu li:nth-child(4)').click();
+              });
+
+              // * Check that confirmation dialog is open.
+              cy.get('.modal-dialog').should('be.visible');
+
```

Comment:
Before step 6 below, please add assertion for "* Confirmation modal warns how many comments will be deleted".

Reply:
I am not sure to understand what do you mean by that. The current modal shows the following text
>Are you sure you want to delete this Post?
What value should I assert then?

Reply:
One way could be comparing the HTML of the modal body where it has number of post.
```javascript
cy.get('#deletePostModalBody').should('be.visible').should('have.html', '<span>Are you sure you want to delete this <span>Post</span>?</span><br><br><span>This post has 1 comment on it.</span>');
```

Note: You may add element ID of `deletePostModalBody` to modal body.